Sources/DistributedCluster/Pattern/WorkerPool.swift

```diff
@@ -53,33 +54,37 @@ public distributed actor WorkerPool<Worker: DistributedWorker>: DistributedWorke
     // MARK: WorkerPool state
 
     /// The worker pool. Worker will be selected round-robin.
-    private var workers: [Worker.ID: Weak<Worker>] = [:]
+    private var workers: OrderedSet<WeakLocalRef<Worker>> = []
     private var roundRobinPos = 0
 
     /// Boolean flag to help determine if pool becomes empty because at least one worker has terminated.
     private var hasTerminatedWorkers = false
     /// Control for waiting and getting notified for new worker.
-    private var newWorkerContinuations: [CheckedContinuation<Void, Never>] = []
+    private var newWorkerContinuations: OrderedSet<ClusterCancellableCheckedContinuation<Void>> = []
 
-    // TODO: remove the convenience marker; since SE-0327 we don't need it anymore: https://github.com/apple/swift-evolution/blob/main/proposals/0327-actor-initializers.md
-    public convenience init(selector: Selector, actorSystem: ActorSystem) async throws {
+    public init(selector: Selector, actorSystem: ActorSystem) async throws {
         try await self.init(settings: .init(selector: selector), actorSystem: actorSystem)
     }
 
     public init(settings: WorkerPoolSettings<Worker>, actorSystem system: ActorSystem) async throws {
         try settings.validate()
 
-        self.actorSystem = system
         self.whenAllWorkersTerminated = settings.whenAllWorkersTerminated
         self.logLevel = settings.logLevel
-
-        switch settings.selector.underlying {
+        self.strategy = settings.strategy
+        self.actorSystem = system
+        
+        switch settings.selector {
         case .dynamic(let key):
             self.newWorkersSubscribeTask = Task {
                 for await worker in await self.actorSystem.receptionist.listing(of: key) {
-                    self.actorSystem.log.log(level: self.logLevel, "Got listing member for \(key): \(worker)")
-                    self.workers[worker.id] = Weak(worker)
+                    self.workers.append(WeakLocalRef(worker))
+                    self.workers.sort(by: { $0.id < $1.id })
```

Comment:
Currently for test to pass I'm sorting workers every time, atm not sure of OrderSet brings any value 🤔
need to double check